src/commands/pin.ts

```diff
@@ -1,94 +1,90 @@
-  },
-});
-
-export async function handlePin(
-  messageToPin: Message | MessageContextMenuCommandInteraction,
-  member: GuildMember,
-) {
-  const msg =
-    messageToPin instanceof ContextMenuCommandInteraction
-      ? messageToPin.targetMessage
-      : messageToPin;
-
-  if (!member.roles.cache.has(env.BEKEND_ROLE_ID))
-    return await messageToPin.reply(
-      "ik ben niet jouw vriend, laat dat ook even duidelijk zijn",
-    );
-
-  if (msg.pinned)
-    return await messageToPin.reply("dit bericht is al gepind aapje");
-
-  await msg.pin();
-  await messageToPin.reply(`bericht gepind door <@${member.id}>!`);
-}
-
-export async function handleUnpin(
-  messageToUnpin: Message | MessageContextMenuCommandInteraction,
-  member: GuildMember,
-) {
-  const msg =
-    messageToUnpin instanceof ContextMenuCommandInteraction
-      ? messageToUnpin.targetMessage
-      : messageToUnpin;
+    if (!message.member.roles.cache.has(env.BEKEND_ROLE_ID)) {
+      message.reply(
+        "ik ben niet jouw vriend, laat dat ook even duidelijk zijn",
+      );
+      return;
+    }
 
-  if (!member.roles.cache.has(env.BEKEND_ROLE_ID))
-    return await messageToUnpin.reply(
-      "ik ben niet jouw vriend, laat dat ook even duidelijk zijn",
-    );
+    const referenced: OmitPartialGroupDMChannel<Message<boolean>> =
+      await message.fetchReference();
 
-  if (!msg.pinned)
-    return await messageToUnpin.reply("dit bericht is niet gepind aapje");
+    if (referenced.pinned) {
+      message.reply(
+        "dit bericht is al gepind aapje",
+      );
+      return;
+    }
 
-  await msg.unpin();
-  await messageToUnpin.reply(`bericht geunpind door <@${member.id}>!`);
-}
+    referenced.pin();
+    message.reply(`bericht gepind door <@${message.author.id}>`);
+  },
+};
 
-registerCommand({
+export const unpin: Command = {
   name: "unpin",
-  command: "unpin",
-  description: "unpin een bericht",
-  handle: async (message, _) => {
+  command: ".unpin",
+  description: "Unpin een bericht",
+  showInHelp: true,
+  match: (message: Message) => message.content === unpin.command,
+  execute: async (message: Message): Promise<void> => {
     if (
-      message.reference == null ||
-      message.reference.messageId == null ||
-      message.author.id === client.user?.id
-    )
-      return await message.reply("omg gebruik dit op een bericht ofz");
-    if (!message.member)
-      return await message.reply(
+      message.reference === null
+      || message.reference.messageId === null
+    ) {
+      message.reply("omg gebruik dit op een bericht ofz");
+      return;
+    }
+
+    if (!message.member) {
+      message.reply(
         "je moet een lid zijn van de server om dit te doen",
       );
+      return;
+    }
 
-    const msg = await message.fetchReference();
+    if (!message.member.roles.cache.has(env.BEKEND_ROLE_ID)) {
+      message.reply(
+        "ik ben niet jouw vriend, laat dat ook even duidelijk zijn",
+      );
+      return;
+    }
```

Comment:
Why did you modify this part?

Reply:
Wat. (Also, ja, dat kan wel, `.s` bestaat. Defensive programming is belangrijk :))